src/coinbase/address/developer_address.ts

```diff
@@ -0,0 +1,364 @@
+import { Decimal } from "decimal.js";
+import { ethers } from "ethers";
+import { Address } from "../address";
+import { Address as AddressModel } from "../../client";
+import { Asset } from "../asset";
+import { Balance } from "../balance";
+import { BalanceMap } from "../balance_map";
+import { Coinbase } from "../coinbase";
+import { ArgumentError, InternalError } from "../errors";
+import { FaucetTransaction } from "../faucet_transaction";
+import { Trade } from "../trade";
+import { Transfer } from "../transfer";
+import { Amount, Destination, TransferStatus } from "../types";
+import { delay } from "../utils";
+import { Wallet as WalletClass } from "../wallet";
+
+/**
+ * A representation of a blockchain address, which is a developer-controlled account on a network.
+ */
+export class DeveloperAddress extends Address {
+  private _model: AddressModel;
+  private _key?: ethers.SigningKey;
+
+  /**
+   * Initializes a new Developer Address instance.
+   *
+   * @param model - The address model data.
+   * @param key - The ethers.js SigningKey the Address uses to sign data.
+   * @throws {InternalError} If the address model is empty.
+   */
+  constructor(model: AddressModel, key?: ethers.SigningKey) {
+    if (!model) {
+      throw new InternalError("Address model cannot be empty");
+    }
+    super(model.network_id, model.address_id);
+
+    this._model = model;
+    this._key = key;
+  }
+
```

Comment:
We will need to have setKey method: https://github.com/coinbase/coinbase-sdk-ruby/pull/86/files#diff-1d829116f97d14de35db53f5a0a130c39e90e5785185b15f9f6efceea7ecf7adR32

Reply:
Just realized I did have a setKet function but somehow was named incorrectly as `getKet` 😅 . Done!